File: binary-search/intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int> set1(nums1.begin(), nums1.end());
        unordered_set<int> res;

        for(int i=0; i<nums2.size();++i){
            if (set1.find(nums2[i])!=set1.end()){
                res.insert(nums2[i]);
            }
        }

        return vector<int>(res.begin(), res.end());
    }
};

File: binary-search/intersection-of-two-arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        res = set()
        num1_set = set()

        for num in nums1:
            num1_set.add(num)

        for num in nums2:
            if num in num1_set:
                res.add(num)
        
        return list(res)